### DISCUSSION TITLE
Cut/Paste/Select/Highlight Text not working in Linux

### DISCUSSION THREAD

**Discussion by donovan** [jesseduffield/lazygit#1667] [Q&A] [answered]
Hi there

Just started using `lazygit` today and I already know I can't go back. That said I have hit a few issues that I would love to try and fix. My issue is that `lazygit` does not behave like any other terminal application I use when it comes to Cutting, Pasting, Selecting and Highlighting text.

For example:
* I can't paste from the clipboard or primary selection (CTRL-SHIFT-V/Middle click) eg in the commit box
* I can't select text in any box to populate the clipboard or primary selection for pasting elsewhere.
* I can't right click on a url to bring up a menu that gives me the option to open link (although the url is underlined as expected).

Is there any way (config option?) I can enable normal behavior? Seems like the feature allowing mouse clicking on boxes to focus them is what is causing this, I don't use that so would happily disable that feature in exchange for working text selection.

I am using `commit=02bf6a5c177c145c46e5e62524458fc3375d01af, build date=2021-11-22T10:08:20Z, build source=binaryRelease, version=0.31.4, os=linux, arch=amd64` in gnome terminal in gnome on Ubuntu 20.04. I have the same behavior across two machines and it was also present in the out of date Ubuntu PPA version.

Thanks

**Comment by donovan** [comment, marked as answer]
Found the solution. For reference:

```
$ cat ~/.config/lazygit/config.yml
gui:
  mouseEvents: false
```

**Reply by T3sT3ro** [reply]
this disables all mouse events like mose wheel etc. I don't think that is the solution.

**Comment by jesseduffield** [comment]
Another option is to hold the option key (assuming OSX) while clicking/dragging so that the mouse events are ignored just for that action

**Reply by prenaux** [reply]
That doesn't work for me in Terminal.app. Is that maybe a iTerm2 only thing?

I'm running lazygit in tmux in a remote ssh in Terminal.app if that makes any difference. Other tmux windows do work with Opt+Click and without it.

**Reply by stefanhaller** [reply]
In Terminal.app it seems to be the `fn` key instead of option. Not sure that's documented anywhere, this is just from cursory testing.

**Reply by ruudk** [reply]
For Ghostty on macOS I have to use `OPTION + SHIFT`.

**Reply by taylorh140** [reply]
for windows (Shift+Alt) works for selecting text.

**Reply by prenaux** [reply]
Thanks to everyone that replied. On Windows in Windows Terminal I can confirm Shift+Alt+Drag does a rectangular selection and Shift+Drag does a per line selection and both work with Lazygit.

There's no "fn" key on my external keyboard so I'm still not sure how to do that on macOS, it works fine with the laptop's keyboard though...

**Reply by stefanhaller** [reply]
> There's no "fn" key on my external keyboard so I'm still not sure how to do that on macOS,

It depends on the terminal app, not the OS. In Terminal.app (which comes with macOS) it is indeed "fn", and you may be out of luck if your keyboard doesn't have one. However, Terminal.app is pretty terrible, so I don't recommend it; other popular terminals are iTerm2 (uses option/alt for selecting text) and Ghostty (uses shift).
